R-package/src/xgboost_R.cc

```diff
@@ -306,18 +306,24 @@ object, need to be destructed before triggering the R error.
 In order to preserve the error message, it gets copied to a temporary
 buffer, and the R error section is reached through a 'goto' statement
 that bypasses usual function control flow. */
-char cpp_ex_msg[512];
+namespace {
+constexpr std::size_t MsgSize = 512;
+char cpp_ex_msg[MsgSize];
```

Comment:
Would this be compliant with standard C++17?

Reply:
No, but I don't know if every standards-compliant compiler out there would accept this pattern (array size determined by an `std::size_t` constexpr variable).

Reply:
Yes, it's valid c++ code. It's used very often